src/pe/export.rs

```diff
@@ -108,6 +108,21 @@ impl<'a> ExportData<'a> {
         let number_of_name_pointers = export_directory_table.number_of_name_pointers as usize;
         let address_table_entries = export_directory_table.address_table_entries as usize;
 
+        if number_of_name_pointers > bytes.len() {
+            let message = format!(
+                "Buffer is too short for {} name pointers",
+                number_of_name_pointers
+            );
+            return Err(error::Error::Malformed(message));
+        }
+        if address_table_entries > bytes.len() {
+            let message = format!(
+                "Buffer is too short for {} address table entries",
+                address_table_entries
+            );
+            return Err(error::Error::Malformed(message));
```

Comment:
I’m wondering if we should introduce a TooShort(count, &’static str) error variant which would do:

“Buffer is too short for {0} {1}” since you write the same boilerplate message every time  (and technically allocate)

Reply:
Good idea! However I noticed that `goblin::Error` is not flagged as `#[non_exhaustive]`, so adding a new variant would be technically a breaking change :-(

So how about doing that as a followup at a later time so it does not block releasing this fix.

Reply:
There is a breaking change pending anyway, so it’s perfect timing.

Reply:
That is good to know, I will do that tomorrow then